### PR TITLE
Update minimum version requirements, add minimum version CI

### PR DESCRIPTION
**Pull request by bhazelton**
## Description

Update the minimum version requirements for several packages based on actual compatibility:
The following packages have increased minimum requirements:
  - python-casacore >= 3.3 (this is an optional dependency)
  - numpy >= 1.19
The following packages had no previous minimum specified:
  - pyyaml >= 5.1 (this is an optional dependency)
  - scipy >= 1.3
  - astropy-healpix >= 0.6 (this is an optional dependency)

Add a CI to run the tests using the minimum versions to ensure compatibility.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation change (documentation changes only)
- [ ] Version change
- [x] Build or continuous integration change

## Checklist:

- [x] I have read the [contribution guide](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/main/.github/CONTRIBUTING.md).
- [x] My code follows the code style of this project.

Documentation change checklist:
- [ ] Any updated docstrings use the [numpy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [ ] If this is a significant change to the readme or other docs, I have checked that they are rendered properly on ReadTheDocs. (you may need help to get this branch to build on RTD, just ask!)

Build or continuous integration change checklist:
- [x] If required or optional dependencies have changed (including version numbers), I have updated the readme to reflect this.
- [ ] If this is a new CI setup, I have added the associated badge to the readme and to references/make_index.py (if appropriate).

### CODE REVIEW

File: ci/pyuvdata_min_versions_tests.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  - astropy==5.0.4
+  - astroquery==0.4.4
+  - h5py==3.1.*
+  - astropy-healpix==0.6
+  - numpy==1.19.*
+  - pyerfa==2.0
+  - python-casacore==3.3.*
+  - pyyaml==5.1.*
+  - scipy==1.3.*
+  - six  # added for now because sometimes an old python-casacore build (before 3.3.1) is used that omits this dependency
```
Comment (mkolopanis) [inline ci/pyuvdata_min_versions_tests.yml:15]: so we tag astropy with a fixed final tick in the version number. Should we do the same with casacore to get rid of this extra baggage? Maybe it's not a big deal in the end, just seems like an easy solution to remove six when we don't need it.

**Review by mkolopanis — COMMENTED** [review]
Looks pretty simple. I do have one question though about casacore.

File: ci/pyuvdata_min_versions_tests.yml (hunk 1 above)
Comment (bhazelton) [reply ci/pyuvdata_min_versions_tests.yml:15]: sure!

**Comment by mkolopanis** [comment]
honestly with how it's basically a bandaid for an astropy issue that is already solved, I'm fine with a pragma on it

**Comment by bhazelton** [comment]
> honestly with how it's basically a bandaid for an astropy issue that is already solved, I'm fine with a pragma on it

Will do. It just feels like giving up 😆

**Comment by mkolopanis** [comment]
I'm pretty sure it is too, but no need to work too hard for this one I feel like.

File: pyuvdata/telescopes.py [hunk 2]
```diff
@@ -174,6 +174,8 @@ def known_telescopes():
         List of known telescope names.
     """
     astropy_sites = EarthLocation.get_site_names()
+    # Empty strings have been removed from astropy sites as of astropy 5.1
+    # This can be removed when we require astropy >= 5.1
     while "" in astropy_sites:
         astropy_sites.remove("")
```
Comment (kartographer) [inline pyuvdata/telescopes.py:180]: What about doing away entirely with the while loop? Something like  this: 
```suggestion
    astropy_sites = [site for site in EarthLocation.get_site_names() if site != ""]
```

Comment (bhazelton) [reply pyuvdata/telescopes.py]: Good idea!

**Review by kartographer — APPROVED** [review]
The dreaded line is covered -- thanks @bhazelton!
